src/java_bytecode/expr2java.cpp

```diff
@@ -491,9 +483,10 @@ std::string expr2javat::convert(
   const exprt &src,
   unsigned &precedence)
 {
+  const typet &type=ns.follow(src.type());
   if(src.id()=="java-this")
```

Comment:
Also unsure why we would need to distinguish the two. I'd defer a potential clean-up of this to a later PR.